unittests/soap_over_udp/test_soap_over_udp.cpp

```diff
@@ -0,0 +1,173 @@
+        qRegisterMetaType<KDSoapMessage>("KDSoapMessage");
+        qRegisterMetaType<KDSoapHeaders>("KDSoapHeaders");
+        qRegisterMetaType<QHostAddress>("QHostAddress");
+    }
+    
+    void testExample() {
+        // This is the example from the KDSoapUdpClient documentation. It is 
+        //   here for compile testing.
+        
+        auto soapUdpClient = new KDSoapUdpClient(this);
+        connect(soapUdpClient, &KDSoapUdpClient::receivedMessage, [=](const KDSoapMessage& message, const KDSoapHeaders& headers, const QHostAddress& address, quint16 port) { 
+            if(message.messageAddressingProperties().action() == QStringLiteral("http://docs.oasis-open.org/ws-dd/ns/discovery/2009/01/ProbeMatches")) {
+                TNS__ProbeMatchesType probeMatches;
+                probeMatches.deserialize(message);
+                qDebug() << "Received probe match from" << address;
+            }
+        });
+        soapUdpClient->bind(3702);
+        
+        TNS__ProbeType probe;
+
+        KDSoapMessage message;
+        message = probe.serialize(QStringLiteral("Probe"));
+        message.setUse(KDSoapMessage::LiteralUse);
+        message.setNamespaceUri(QStringLiteral("http://docs.oasis-open.org/ws-dd/ns/discovery/2009/01"));
+
+        KDSoapMessageAddressingProperties addressing;
+        addressing.setAction(QStringLiteral("http://docs.oasis-open.org/ws-dd/ns/discovery/2009/01/Probe"));
+        addressing.setMessageID(QStringLiteral("urn:uuid:") + QUuid::createUuid().toString(QUuid::WithoutBraces));
+        addressing.setDestination(QStringLiteral("urn:docs-oasis-open-org:ws-dd:ns:discovery:2009:01"));
+        addressing.setReplyEndpointAddress(KDSoapMessageAddressingProperties::predefinedAddressToString(KDSoapMessageAddressingProperties::Anonymous));
+        message.setMessageAddressingProperties(addressing);
+
+        soapUdpClient->sendMessage(message, KDSoapHeaders(), QHostAddress("239.255.255.250"), 3702);
+    }
+    
+    void testSendMessage() {
+        QUdpSocket testSocket;
+        bool rc = testSocket.bind();
+        Q_ASSERT(rc);
+        
+        KDSoapUdpClient udpClient;
+        
+        udpClient.sendMessage(exampleMessage(), KDSoapHeaders(), QHostAddress::LocalHost, testSocket.localPort());
+        
+        QVERIFY(testSocket.hasPendingDatagrams());
+        auto datagram = testSocket.receiveDatagram();
+        
+        QVERIFY(KDSoapUnitTestHelpers::xmlBufferCompare(datagram.data(), exampleTextData()));
+    }
+    
+    void testReceiveMessage() {
+        QUdpSocket testSocket;
+        
+        KDSoapUdpClient udpClient;
+        bool rc = udpClient.bind(12345);
+        QVERIFY(rc);
+        
+        QSignalSpy spy(&udpClient, SIGNAL(receivedMessage(KDSoapMessage, KDSoapHeaders, QHostAddress, quint16)));
+        
+        auto data = exampleTextData();
+        qint64 size = testSocket.writeDatagram(data, QHostAddress::LocalHost, 12345);
+        QVERIFY(size == data.size());
```

Comment:
QCOMPARE (so that we can see values in case of failure)

Reply:
Makes sense